### DISCUSSION TITLE
HMS 350 openDTU with Esp32,cmt2300

### DISCUSSION THREAD

**Discussion by RS2480** [tbnobody/OpenDTU#2188] [General]
Hello everyone, I need a help i have used HMS350 to use with opendtu with esp-wroom-32 and cmt 2300. I followed the steps on the opendtu documentation but I am not getting the live data of the inverter on the opendtu platform.

my pin mapping file looks like this

 [
    {

        "name": "Generic NodeMCU 38 pin",
     "Cmt": {
            "clk": 12,
            "cs": 27,
            "fcs": 26,
            "sdio": 14,
            "gpio2": -1,
            "gpio3": -1
        },

       "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }    
      }
]

**Comment by stefan123t** [comment]
@RS2480 can you send us the first seven digits of your Inverters Serial ID. We need that to identify the Model type and manufacturer year and calendar week.

P lease double check your model type too, there are inverters sold with an built-in DTUBI which is the Hoymiles WiFi Series models known as eg HMS-350**W** these do not work with OpenDTU/Ahoy. Please refer eg to the project from DennisOSRM for an option to eg integrate one HMS-xxx**W** inverter into home assistant.

**Reply by RS2480** [reply]
Thanks for your reply, i have checked my inverter is HMS-350-1T  SN:1124905xxxxx.  Could you send me the steps in short so i can refer that for opendtu.
